server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -423,43 +434,71 @@ private void snapshot(
-                            + "for index "
-                            + indexId.getName()
-                            + ", shard "
-                            + shardId.getId()
-                            + " and snapshot "
-                            + snapshot.getSnapshotId()
-                            + " is "
-                            + (endTime - startTime)
+                    GatedCloseable<IndexCommit> finalWrappedSnapshot = wrappedSnapshot;
+                    long finalCommitGeneration = commitGeneration;
+                    repository.snapshotRemoteStoreIndexShard(
+                        indexShard.store(),
+                        snapshot.getSnapshotId(),
+                        indexId,
+                        snapshotIndexCommit,
+                        getShardStateId(indexShard, snapshotIndexCommit),
+                        snapshotStatus,
+                        primaryTerm,
+                        startTime,
+                        new ActionListener<String>() {
+                            @Override
+                            public void onResponse(String s) {
+                                long endTime = threadPool.relativeTimeInMillis();
+                                logger.debug(
+                                    "Time taken (in milliseconds) to complete shallow copy snapshot, "
+                                        + "for index "
+                                        + indexId.getName()
+                                        + ", shard "
+                                        + shardId.getId()
+                                        + " and snapshot "
+                                        + snapshot.getSnapshotId()
+                                        + " is "
+                                        + (endTime - startTime)
+                                );
+                                try {
+                                    finalWrappedSnapshot.close();
+                                } catch (IOException e) {
+                                    listener.onFailure(e);
+                                }
+                                listener.onResponse(s);
+                            }
+
+                            @Override
+                            public void onFailure(Exception e) {
+                                logger.error(
+                                    "Shallow Copy Snapshot Failed for Shard ["
+                                        + indexId.getName()
+                                        + "]["
+                                        + shardId.getId()
+                                        + "] for snapshot "
+                                        + snapshot.getSnapshotId()
+                                        + ", releasing acquired lock from remote store"
+                                );
+
+                                try {
+                                    indexShard.releaseLockOnCommitData(
```

Comment:
What happens if lock was not acquired? Does this become a no-op?

Reply:
yes, release lock will be no-op if md file or lock file does not exist.